tasks/actorstate/miner/sector_events.go

```diff
@@ -25,27 +26,27 @@ func (SectorEventsExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
 		span.SetAttributes(a.Attributes()...)
 	}
 
-	ec, err := NewMinerStateExtractionContext(ctx, a, node)
+	ec, err := extraction.LoadMinerStates(ctx, a, node)
 	if err != nil {
 		return nil, fmt.Errorf("creating miner state extraction context: %w", err)
 	}
 
 	var sectorChanges *miner.SectorChanges
 	var preCommitChanges *miner.PreCommitChanges
-	if !ec.HasPreviousState() {
+	if ec.ParentState() == nil {
 		// If the miner doesn't have previous state list all of its current sectors and precommits
-		sectors, err := ec.CurrState.LoadSectors(nil)
+		sectors, err := ec.CurrentState().LoadSectors(nil)
 		if err != nil {
 			return nil, fmt.Errorf("loading miner sectors: %w", err)
 		}
 
 		sectorChanges = miner.MakeSectorChanges()
-		for i, sector := range sectors {
-			sectorChanges.Added[i] = *sector
+		for _, sector := range sectors {
+			sectorChanges.Added = append(sectorChanges.Added, *sector)
```

Comment:
Fixes a panic as we need to append to `Added`. This is a rare case when processing the genesis block.